pandas/core/groupby/generic.py

```diff
@@ -1061,6 +1061,9 @@ def _cython_agg_blocks(
                 else:
                     result = cast(DataFrame, result)
                     # unwrap DataFrame to get array
+                    if len(result._data.blocks) != 1:
```

Comment:
what? this is pretty odd to do

if there is an error this is not the way to fix it

Reply:
i think it must be that we're going through Block._split_and_operate; can anyone think of another way this would happen?